server/middleware/registration/index.js

```diff
@@ -0,0 +1,126 @@
+const bcrypt = require('bcrypt');
+const {
+  HttpStatusCodes: { StatusCodes },
+  DataException
+} = require('../../utils/constants');
+const { REQUEST_TIMEOUT, OK, BAD_REQUEST, GATEWAY_TIMEOUT } = StatusCodes;
+const mongoUtil = require('../../utils/mongoUtil');
+const JWT = require('jsonwebtoken');
+const { v4 } = require('uuid');
+const registrationLogger = require('./registrationLogger');
+
+const resObj = (statusCode, message, data = {}) => ({
+  statusCode,
+  message,
+  data,
+});
+
+const isFormValid = ({ body }) => {
+  const formFields = [
+    // 'role_id',
+    'firstName',
+    'lastName',
+    'title',
+    'email',
+    'password'
+  ];
+  
+  for (let i = 0; i < formFields.length; i++) {
+    const field = formFields[i];
+    if (!Object.hasOwnProperty.call(body, formFields[i]) || body[field].length === 0) {
+      return false;
+    }
+  }
+
+  const emailPattern = /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
+  if (!emailPattern.test(body.email)) {
+    return false;
+  } else {
+    return true;
+  }
+
+};
+
+const hashForm = ({ body }) => {
+  const saltRounds = 10;
+  body.email = body.email.toLowerCase();
+  const { password, ...rest } = body;
```

Comment:
pick only the payload you're going to save in the storage, ...rest will include retypePassword, we don't want the plain text version saved in the db

`const { password, email, title, role_id, lastName, firstName } = body;`